### PR TITLE
Pinning specific actions versions

### LINKED ISSUE #615: update deprecated Actions
see for example https://github.com/mandiant/flare-floss/actions/runs/3836568797
`checkout` and `setup-python`

### PR DESCRIPTION
**Pull request by ooprathamm**
Fixes #615
Pinning specific actions versions and Adding comment to explain the reason.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ jobs:
           - os: macos-11
             artifact_name: floss
             asset_name: macos
+    # Pinning actions version to avoid Node.js depreciation warnings
```
[the remark this answers is not in the document]
Comment (williballenthin) [reply .github/workflows/build.yml:26]: and please use this comment throughout

[the remark this answers is not in the document]
Comment (williballenthin) [reply .github/workflows/build.yml:26]: but only need to use it once per file

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -10,10 +10,11 @@ on:
 jobs:
   deploy:
     runs-on: ubuntu-20.04
+    # Pinning actions version to avoid Node.js depreciation warnings
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
       - name: Set up Python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435 # v4.5.0
```
Comment (williballenthin) [inline .github/workflows/publish.yml:17]: looks good, thanks!

**Review by williballenthin — CHANGES_REQUESTED** [review]
small nit on the comment text. once fixed, happy to merge!

**Comment by ooprathamm** [comment]
@williballenthin made the changes.

**Review by williballenthin — APPROVED** [review]
thank you!

**Pull request merged**